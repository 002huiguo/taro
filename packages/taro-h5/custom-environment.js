const PuppeteerEnvironment = require('jest-environment-puppeteer')

/**
 * For unit test debug
 */
class CustomEnvironment extends PuppeteerEnvironment {
  async setup () {
    await super.setup()
    // Your setup
  }

  async teardown () {
    // await new Promise((resolve, reject) => {
    //   setTimeout(() => {resolve(1)}, 20000)
    // })
    // Your teardown
    await super.teardown()
  }
}

module.exports = CustomEnvironment
