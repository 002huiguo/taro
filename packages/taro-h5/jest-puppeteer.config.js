module.exports = {
  launch: {
    headless: false
  },
  server: {
    command: 'npm run test-server',
    port: 9000,
    launchTimeout: 10000
  }
}
