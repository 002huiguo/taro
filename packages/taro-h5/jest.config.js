module.exports = {
  testMatch: ['<rootDir>/__test__/**/*-test.js'],
  setupFiles: ['jest-localstorage-mock'],
  setupTestFrameworkScriptFile: 'jest-mock-console/dist/setupTestFramework.js'
}
