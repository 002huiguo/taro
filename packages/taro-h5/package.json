{
  "name": "@tarojs/taro-h5",
  "version": "1.2.17",
  "description": "Taro h5 framework",
  "main:h5": "src/index.js",
  "main": "dist/index.js",
  "typings": "types/index.d.ts",
  "files": [
    "dist",
    "src",
    "index.js",
    "types"
  ],
  "sideEffects": [
    "./src/taro/index.js"
  ],
  "scripts": {
    "test": "jest",
    "test:dev": "jest --watch",
    "test:coverage": "jest --coverage",
    "build": "rollup -c",
    "dev": "rollup -cw"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/NervJS/taro.git"
  },
  "keywords": [
    "taro"
  ],
  "author": "O2Team",
  "license": "MIT",
  "dependencies": {
    "@tarojs/taro": "1.2.17",
    "base64-js": "^1.3.0",
    "jsonp-retry": "^1.0.3",
    "lodash": "^4.17.11",
    "mobile-detect": "^1.4.2",
    "rollup": "^1.1.2",
    "whatwg-fetch": "^2.0.4"
  },
  "devDependencies": {
    "@babel/node": "^7.2.2",
    "@babel/plugin-proposal-class-properties": "^7.3.0",
    "@babel/plugin-proposal-object-rest-spread": "^7.3.1",
    "@babel/plugin-transform-react-jsx": "^7.3.0",
    "@babel/preset-env": "^7.3.1",
    "babel-core": "^6.26.3",
    "babel-jest": "^23.0.1",
    "babel-preset-env": "^1.7.0",
    "jest-fetch-mock": "^1.6.5",
    "jest-localstorage-mock": "^2.2.0",
    "jest-mock-console": "^0.3.5",
    "mock-socket": "^7.1.0",
    "nervjs": "^1.3.0",
    "rollup": "^1.1.2",
    "rollup-plugin-alias": "^1.5.1",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.2.0",
    "rollup-plugin-node-resolve": "^4.0.0",
    "rollup-pluginutils": "^2.3.3"
  }
}
