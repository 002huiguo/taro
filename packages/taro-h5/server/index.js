import Taro from '../src'
Taro.initNativeApi(Taro)

window.Taro = Taro
