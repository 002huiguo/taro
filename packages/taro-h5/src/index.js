/* eslint-disable camelcase */
import { getEnv, ENV_TYPE, internal_safe_get, internal_dynamic_recursive } from '@tarojs/taro'

import initNativeApi from './native-api'

export {
  initNativeApi,
  getEnv,
  ENV_TYPE,
  internal_safe_get,
  internal_dynamic_recursive
}

export default {
  initNativeApi,
  getEnv,
  ENV_TYPE,
  internal_safe_get,
  internal_dynamic_recursive
}
