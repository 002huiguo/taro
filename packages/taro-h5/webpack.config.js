const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  entry: './server/index.js',
  resolve: {
    extensions: ['.js']
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        loaders: ['babel-loader']
      }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      filename: 'index.html',
      template: './server/index.html'
    })
  ],
  devServer: {
    port: 9000
  }
}
