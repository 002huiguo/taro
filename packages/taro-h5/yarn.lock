# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@tarojs/taro@0.0.31":
  version "0.0.31"
  resolved "https://registry.npmjs.org/@tarojs/taro/-/taro-0.0.31.tgz#f517e4f0dcf480e40472ca7ccdb3956340330003"

jsonp-retry@^1.0.3:
  version "1.0.3"
  resolved "https://registry.npmjs.org/jsonp-retry/-/jsonp-retry-1.0.3.tgz#9e6b18b02ec767a621dd44b4e99439a8a43a6419"
  dependencies:
    object-assign "^4.1.1"

object-assign@^4.1.1:
  version "4.1.1"
  resolved "https://registry.npmjs.org/object-assign/-/object-assign-4.1.1.tgz#2109adc7965887cfc05cbbd442cac8bfbb360863"
