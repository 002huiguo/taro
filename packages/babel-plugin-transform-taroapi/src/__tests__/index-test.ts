import * as apis from '@tarojs/taro-h5/dist/taroApis'
import * as babel from 'babel-core'
import * as t from 'babel-types'

import plugin from '../'

const pluginOptions = [
  plugin,
  {
    apis,
    packageName: '@tarojs/taro-h5'
  }
]

const getNamedImports = (importSpecifiers: (t.ImportSpecifier | t.ImportDefaultSpecifier | t.ImportNamespaceSpecifier)[]) => {
  return importSpecifiers.reduce((prev, curr: t.ImportSpecifier) => {
    if (t.isImportSpecifier(curr)) {
      prev.add(curr.imported.name)
    }
    return prev
  }, new Set())
}

it('should work!', function () {
  const code = `
    import Taro, { setStorage, initPxTransform, param } from '@tarojs/taro-h5';
    initPxTransform(param)
    Taro.initPxTransform()
    Taro.initPxTransform()
    Taro['getStorage']()
    setStorage()
    export { Taro }
  `
  const result = babel.transform(code, {plugins: [pluginOptions]});
  expect(result.code).toMatchSnapshot();
});

it('should transform fullfilled apis to named import', function () {
  const code = `
    import Taro from '@tarojs/taro-h5'
    ${Array.from(apis).map(api => `Taro.${api}();`)
    .join('')}
  `
  const result = babel.transform(code, { plugins: [pluginOptions] })
  expect(result.code).toMatchSnapshot();

  const ast = result.ast as t.File
  const body = ast.program.body as [t.ImportDeclaration, t.ExpressionStatement]
  expect(t.isImportDeclaration(body[0])).toBeTruthy()

  const namedImports = getNamedImports(body[0].specifiers)
  expect(namedImports).toEqual(new Set(apis))
})

it('should leave other apis untouched', function () {
  const code = `
    import Taro from '@tarojs/taro-h5'
    Taro.noop
  `
  const result = babel.transform(code, { plugins: [pluginOptions] })
  expect(result.code).toMatchSnapshot();

  const ast = result.ast as t.File
  const body = ast.program.body as [t.ImportDeclaration, t.ExpressionStatement]
  expect(t.isImportDeclaration(body[0])).toBeTruthy()
  expect(t.isExpressionStatement(body[1])).toBeTruthy()
  const defaultImport = body[0].specifiers.find(v => t.isImportDefaultSpecifier(v))
  expect(defaultImport).toBeTruthy()

  const taroName = defaultImport!.local.name
  const namedImports = getNamedImports(body[0].specifiers)
  expect(namedImports).toEqual(new Set())
  expect(t.isMemberExpression(body[1].expression)).toBeTruthy()
  expect((body[1].expression as t.MemberExpression)).toMatchObject(t.memberExpression(
    t.identifier(taroName),
    t.identifier('noop')
  ))
})

it('should move static apis under "Taro"', function () {
  const code = `
    import { noop } from '@tarojs/taro-h5';
    noop;
    noop();
  `

  const result = babel.transform(code, { plugins: [pluginOptions] })
  expect(result.code).toMatchSnapshot();

  const ast = result.ast as t.File
  const body = ast.program.body as [t.ImportDeclaration, t.ExpressionStatement]
  expect(t.isImportDeclaration(body[0])).toBeTruthy()
  expect(t.isExpressionStatement(body[1])).toBeTruthy()
  const defaultImport = body[0].specifiers.find(v => t.isImportDefaultSpecifier(v))
  expect(defaultImport).toBeTruthy()

  const taroName = defaultImport!.local.name
  let memberExpression = body[1].expression
  if (t.isCallExpression(body[1])) {
    memberExpression = (body[1].expression as t.CallExpression).callee
  }
  expect(memberExpression).toMatchObject(t.memberExpression(
    t.identifier(taroName),
    t.identifier('noop')
  ))
})

it('should not import taro duplicatly', function () {
  const code = `
    import { Component } from "@tarojs/taro-h5";
    import Taro from '@tarojs/taro-h5';
    Component
    Taro.createAnimation()
    Taro.initPxTransform()
  `

  const result = babel.transform(code, { plugins: [pluginOptions] })
  expect(result.code).toMatchSnapshot();

  const ast = result.ast as t.File
  const body = ast.program.body as [t.ImportDeclaration, t.ExpressionStatement, t.ExpressionStatement]
  expect(t.isImportDeclaration(body[0])).toBeTruthy()
  expect(t.isExpressionStatement(body[1])).toBeTruthy()
  expect(t.isExpressionStatement(body[2])).toBeTruthy()
})

it('should not go wrong when use an api twice', function () {
  const code = `
    import Taro from '@tarojs/taro-h5';
    const animation = Taro.createAnimation({
      duration: dura * 1000,
      timingFunction: 'linear'
    })
    const resetAnimation = Taro.createAnimation({
      duration: 0,
      timingFunction: 'linear'
    })
  `
  expect(() => {
    const result = babel.transform(code, { plugins: [ pluginOptions ]})
    expect(result.code).toMatchSnapshot();
  }).not.toThrowError()
})