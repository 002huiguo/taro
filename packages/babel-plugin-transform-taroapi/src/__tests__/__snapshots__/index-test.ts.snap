// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should leave other apis untouched 1`] = `
"
import Taro from '@tarojs/taro-h5';
Taro.noop;"
`;

exports[`should move static apis under "Taro" 1`] = `
"
import Taro from '@tarojs/taro-h5';
Taro.noop;
Taro.noop();"
`;

exports[`should not go wrong when using an api twice 1`] = `
"
import { createAnimation as _createAnimation } from '@tarojs/taro-h5';
const animation = _createAnimation({
  duration: dura * 1000,
  timingFunction: 'linear'
});
const resetAnimation = _createAnimation({
  duration: 0,
  timingFunction: 'linear'
});"
`;

exports[`should not import taro duplicatly 1`] = `
"
import Taro, { createAnimation as _createAnimation } from \\"@tarojs/taro-h5\\";

Taro.Component;
_createAnimation();
Taro.initPxTransform();"
`;

exports[`should transform fullfilled apis to named import 1`] = `
"
import { onBackgroundAudioPlay as _onBackgroundAudioPlay, onBackgroundAudioPause as _onBackgroundAudioPause, onBackgroundAudioStop as _onBackgroundAudioStop, onAccelerometerChange as _onAccelerometerChange, onCompassChange as _onCompassChange, onBluetoothAdapterStateChange as _onBluetoothAdapterStateChange, onBluetoothDeviceFound as _onBluetoothDeviceFound, onBLEConnectionStateChange as _onBLEConnectionStateChange, onBLECharacteristicValueChange as _onBLECharacteristicValueChange, onBeaconUpdate as _onBeaconUpdate, onBeaconServiceChange as _onBeaconServiceChange, onUserCaptureScreen as _onUserCaptureScreen, onHCEMessage as _onHCEMessage, onGetWifiList as _onGetWifiList, onWifiConnected as _onWifiConnected, getExtConfigSync as _getExtConfigSync, getLogManager as _getLogManager, onMemoryWarning as _onMemoryWarning, reportAnalytics as _reportAnalytics, navigateToSmartGameProgram as _navigateToSmartGameProgram, getFileSystemManager as _getFileSystemManager, stopRecord as _stopRecord, getRecorderManager as _getRecorderManager, pauseVoice as _pauseVoice, stopVoice as _stopVoice, pauseBackgroundAudio as _pauseBackgroundAudio, stopBackgroundAudio as _stopBackgroundAudio, getBackgroundAudioManager as _getBackgroundAudioManager, createAudioContext as _createAudioContext, createInnerAudioContext as _createInnerAudioContext, createVideoContext as _createVideoContext, createCameraContext as _createCameraContext, createLivePlayerContext as _createLivePlayerContext, createLivePusherContext as _createLivePusherContext, createMapContext as _createMapContext, canIUse as _canIUse, startAccelerometer as _startAccelerometer, stopAccelerometer as _stopAccelerometer, startCompass as _startCompass, stopCompass as _stopCompass, showNavigationBarLoading as _showNavigationBarLoading, hideNavigationBarLoading as _hideNavigationBarLoading, pageScrollTo as _pageScrollTo, createCanvasContext as _createCanvasContext, createContext as _createContext, drawCanvas as _drawCanvas, hideKeyboard as _hideKeyboard, stopPullDownRefresh as _stopPullDownRefresh, createIntersectionObserver as _createIntersectionObserver, onWindowResize as _onWindowResize, offWindowResize as _offWindowResize, getAccountInfoSync as _getAccountInfoSync, getUpdateManager as _getUpdateManager, createWorker as _createWorker, uploadFile as _uploadFile, downloadFile as _downloadFile, previewImage as _previewImage, getImageInfo as _getImageInfo, saveImageToPhotosAlbum as _saveImageToPhotosAlbum, startRecord as _startRecord, playVoice as _playVoice, setInnerAudioOption as _setInnerAudioOption, getAvailableAudioSources as _getAvailableAudioSources, getBackgroundAudioPlayerState as _getBackgroundAudioPlayerState, playBackgroundAudio as _playBackgroundAudio, seekBackgroundAudio as _seekBackgroundAudio, chooseVideo as _chooseVideo, saveVideoToPhotosAlbum as _saveVideoToPhotosAlbum, loadFontFace as _loadFontFace, saveFile as _saveFile, getFileInfo as _getFileInfo, getSavedFileList as _getSavedFileList, getSavedFileInfo as _getSavedFileInfo, removeSavedFile as _removeSavedFile, openDocument as _openDocument, switchTab as _switchTab, reLaunch as _reLaunch, getLocation as _getLocation, chooseLocation as _chooseLocation, openLocation as _openLocation, scanCode as _scanCode, setClipboardData as _setClipboardData, getClipboardData as _getClipboardData, openBluetoothAdapter as _openBluetoothAdapter, closeBluetoothAdapter as _closeBluetoothAdapter, getBluetoothAdapterState as _getBluetoothAdapterState, startBluetoothDevicesDiscovery as _startBluetoothDevicesDiscovery, stopBluetoothDevicesDiscovery as _stopBluetoothDevicesDiscovery, getBluetoothDevices as _getBluetoothDevices, getConnectedBluetoothDevices as _getConnectedBluetoothDevices, createBLEConnection as _createBLEConnection, closeBLEConnection as _closeBLEConnection, getBLEDeviceServices as _getBLEDeviceServices, getBLEDeviceCharacteristics as _getBLEDeviceCharacteristics, readBLECharacteristicValue as _readBLECharacteristicValue, writeBLECharacteristicValue as _writeBLECharacteristicValue, notifyBLECharacteristicValueChange as _notifyBLECharacteristicValueChange, startBeaconDiscovery as _startBeaconDiscovery, stopBeaconDiscovery as _stopBeaconDiscovery, getBeacons as _getBeacons, setScreenBrightness as _setScreenBrightness, getScreenBrightness as _getScreenBrightness, setKeepScreenOn as _setKeepScreenOn, vibrateLong as _vibrateLong, vibrateShort as _vibrateShort, addPhoneContact as _addPhoneContact, getHCEState as _getHCEState, startHCE as _startHCE, stopHCE as _stopHCE, sendHCEMessage as _sendHCEMessage, startWifi as _startWifi, stopWifi as _stopWifi, connectWifi as _connectWifi, getWifiList as _getWifiList, setWifiList as _setWifiList, getConnectedWifi as _getConnectedWifi, setNavigationBarColor as _setNavigationBarColor, setTopBarText as _setTopBarText, startPullDownRefresh as _startPullDownRefresh, canvasToTempFilePath as _canvasToTempFilePath, canvasGetImageData as _canvasGetImageData, canvasPutImageData as _canvasPutImageData, setBackgroundColor as _setBackgroundColor, setBackgroundTextStyle as _setBackgroundTextStyle, getExtConfig as _getExtConfig, login as _login, checkSession as _checkSession, authorize as _authorize, getUserInfo as _getUserInfo, checkIsSupportFacialRecognition as _checkIsSupportFacialRecognition, startFacialRecognitionVerify as _startFacialRecognitionVerify, startFacialRecognitionVerifyAndUploadVideo as _startFacialRecognitionVerifyAndUploadVideo, faceVerifyForPay as _faceVerifyForPay, requestPayment as _requestPayment, showShareMenu as _showShareMenu, hideShareMenu as _hideShareMenu, updateShareMenu as _updateShareMenu, getShareInfo as _getShareInfo, chooseAddress as _chooseAddress, addCard as _addCard, openCard as _openCard, openSetting as _openSetting, getSetting as _getSetting, getWeRunData as _getWeRunData, navigateToMiniProgram as _navigateToMiniProgram, navigateBackMiniProgram as _navigateBackMiniProgram, chooseInvoice as _chooseInvoice, chooseInvoiceTitle as _chooseInvoiceTitle, checkIsSupportSoterAuthentication as _checkIsSupportSoterAuthentication, startSoterAuthentication as _startSoterAuthentication, checkIsSoterEnrolledInDevice as _checkIsSoterEnrolledInDevice, setEnableDebug as _setEnableDebug, ocrIdCard as _ocrIdCard, ocrBankCard as _ocrBankCard, ocrDrivingLicense as _ocrDrivingLicense, ocrVehicleLicense as _ocrVehicleLicense, textReview as _textReview, textToAudio as _textToAudio, imageAudit as _imageAudit, advancedGeneralIdentify as _advancedGeneralIdentify, objectDetectIdentify as _objectDetectIdentify, carClassify as _carClassify, dishClassify as _dishClassify, logoClassify as _logoClassify, animalClassify as _animalClassify, plantClassify as _plantClassify, getSwanId as _getSwanId, requestPolymerPayment as _requestPolymerPayment, navigateToSmartProgram as _navigateToSmartProgram, navigateBackSmartProgram as _navigateBackSmartProgram, preloadSubPackage as _preloadSubPackage, request as _request, createAnimation as _createAnimation, createSelectorQuery as _createSelectorQuery, connectSocket as _connectSocket, onSocketOpen as _onSocketOpen, onSocketError as _onSocketError, sendSocketMessage as _sendSocketMessage, onSocketMessage as _onSocketMessage, closeSocket as _closeSocket, onSocketClose as _onSocketClose, setStorage as _setStorage, setStorageSync as _setStorageSync, getStorage as _getStorage, getStorageSync as _getStorageSync, getStorageInfo as _getStorageInfo, getStorageInfoSync as _getStorageInfoSync, removeStorage as _removeStorage, removeStorageSync as _removeStorageSync, clearStorage as _clearStorage, clearStorageSync as _clearStorageSync, showToast as _showToast, hideToast as _hideToast, showLoading as _showLoading, hideLoading as _hideLoading, showModal as _showModal, showActionSheet as _showActionSheet, initTabBarApis as _initTabBarApis, setTabBarBadge as _setTabBarBadge, removeTabBarBadge as _removeTabBarBadge, showTabBarRedDot as _showTabBarRedDot, hideTabBarRedDot as _hideTabBarRedDot, showTabBar as _showTabBar, hideTabBar as _hideTabBar, setTabBarStyle as _setTabBarStyle, setTabBarItem as _setTabBarItem, getSystemInfo as _getSystemInfo, getSystemInfoSync as _getSystemInfoSync, getNetworkType as _getNetworkType, onNetworkStatusChange as _onNetworkStatusChange, arrayBufferToBase64 as _arrayBufferToBase, base64ToArrayBuffer as _base64ToArrayBuffer, makePhoneCall as _makePhoneCall, setNavigationBarTitle as _setNavigationBarTitle } from '@tarojs/taro-h5';
_onBackgroundAudioPlay();_onBackgroundAudioPause();_onBackgroundAudioStop();_onAccelerometerChange();_onCompassChange();_onBluetoothAdapterStateChange();_onBluetoothDeviceFound();_onBLEConnectionStateChange();_onBLECharacteristicValueChange();_onBeaconUpdate();_onBeaconServiceChange();_onUserCaptureScreen();_onHCEMessage();_onGetWifiList();_onWifiConnected();_getExtConfigSync();_getLogManager();_onMemoryWarning();_reportAnalytics();_navigateToSmartGameProgram();_getFileSystemManager();_stopRecord();_getRecorderManager();_pauseVoice();_stopVoice();_pauseBackgroundAudio();_stopBackgroundAudio();_getBackgroundAudioManager();_createAudioContext();_createInnerAudioContext();_createVideoContext();_createCameraContext();_createLivePlayerContext();_createLivePusherContext();_createMapContext();_canIUse();_startAccelerometer();_stopAccelerometer();_startCompass();_stopCompass();_showNavigationBarLoading();_hideNavigationBarLoading();_pageScrollTo();_createCanvasContext();_createContext();_drawCanvas();_hideKeyboard();_stopPullDownRefresh();_createIntersectionObserver();_onWindowResize();_offWindowResize();_getAccountInfoSync();_getUpdateManager();_createWorker();_uploadFile();_downloadFile();_previewImage();_getImageInfo();_saveImageToPhotosAlbum();_startRecord();_playVoice();_setInnerAudioOption();_getAvailableAudioSources();_getBackgroundAudioPlayerState();_playBackgroundAudio();_seekBackgroundAudio();_chooseVideo();_saveVideoToPhotosAlbum();_loadFontFace();_saveFile();_getFileInfo();_getSavedFileList();_getSavedFileInfo();_removeSavedFile();_openDocument();_switchTab();_reLaunch();_getLocation();_chooseLocation();_openLocation();_scanCode();_setClipboardData();_getClipboardData();_openBluetoothAdapter();_closeBluetoothAdapter();_getBluetoothAdapterState();_startBluetoothDevicesDiscovery();_stopBluetoothDevicesDiscovery();_getBluetoothDevices();_getConnectedBluetoothDevices();_createBLEConnection();_closeBLEConnection();_getBLEDeviceServices();_getBLEDeviceCharacteristics();_readBLECharacteristicValue();_writeBLECharacteristicValue();_notifyBLECharacteristicValueChange();_startBeaconDiscovery();_stopBeaconDiscovery();_getBeacons();_setScreenBrightness();_getScreenBrightness();_setKeepScreenOn();_vibrateLong();_vibrateShort();_addPhoneContact();_getHCEState();_startHCE();_stopHCE();_sendHCEMessage();_startWifi();_stopWifi();_connectWifi();_getWifiList();_setWifiList();_getConnectedWifi();_setNavigationBarColor();_setTopBarText();_startPullDownRefresh();_canvasToTempFilePath();_canvasGetImageData();_canvasPutImageData();_setBackgroundColor();_setBackgroundTextStyle();_getExtConfig();_login();_checkSession();_authorize();_getUserInfo();_checkIsSupportFacialRecognition();_startFacialRecognitionVerify();_startFacialRecognitionVerifyAndUploadVideo();_faceVerifyForPay();_requestPayment();_showShareMenu();_hideShareMenu();_updateShareMenu();_getShareInfo();_chooseAddress();_addCard();_openCard();_openSetting();_getSetting();_getWeRunData();_navigateToMiniProgram();_navigateBackMiniProgram();_chooseInvoice();_chooseInvoiceTitle();_checkIsSupportSoterAuthentication();_startSoterAuthentication();_checkIsSoterEnrolledInDevice();_setEnableDebug();_ocrIdCard();_ocrBankCard();_ocrDrivingLicense();_ocrVehicleLicense();_textReview();_textToAudio();_imageAudit();_advancedGeneralIdentify();_objectDetectIdentify();_carClassify();_dishClassify();_logoClassify();_animalClassify();_plantClassify();_getSwanId();_requestPolymerPayment();_navigateToSmartProgram();_navigateBackSmartProgram();_preloadSubPackage();_request();_createAnimation();_createSelectorQuery();_connectSocket();_onSocketOpen();_onSocketError();_sendSocketMessage();_onSocketMessage();_closeSocket();_onSocketClose();_setStorage();_setStorageSync();_getStorage();_getStorageSync();_getStorageInfo();_getStorageInfoSync();_removeStorage();_removeStorageSync();_clearStorage();_clearStorageSync();_showToast();_hideToast();_showLoading();_hideLoading();_showModal();_showActionSheet();_initTabBarApis();_setTabBarBadge();_removeTabBarBadge();_showTabBarRedDot();_hideTabBarRedDot();_showTabBar();_hideTabBar();_setTabBarStyle();_setTabBarItem();_getSystemInfo();_getSystemInfoSync();_getNetworkType();_onNetworkStatusChange();_arrayBufferToBase();_base64ToArrayBuffer();_makePhoneCall();_setNavigationBarTitle();"
`;

exports[`should work! 1`] = `
"
import Taro, { setStorage as _setStorage, getStorage as _getStorage } from '@tarojs/taro-h5';
Taro.initPxTransform(Taro.param);
Taro.initPxTransform();
Taro.initPxTransform();
_getStorage();
_setStorage();
export { Taro };"
`;
